### PR TITLE
Re-add content length, semi revert of changes in #1010

### LINKED ISSUE #1267: 10.3 breaks skip forward/backward
I've observed a similar problem back to 10.3 rc1
and sadly the newly released 10.3 final release does the same.
Skipping ahead (either by cursor or using my magic remote's cursor) causes the video to restart from the beginning.
Reverting to 10.2.2 fixes the problem.

Thanks in advance,

### PR DESCRIPTION
**Pull request by cvium**
Fixes #1267

### CODE REVIEW

File: Emby.Server.Implementations/HttpServer/FileWriter.cs [hunk 1]
```diff
@@ -67,6 +67,7 @@ public FileWriter(string path, string contentType, string rangeHeader, ILogger l
 
             if (string.IsNullOrWhiteSpace(rangeHeader))
             {
+                Headers[HeaderNames.ContentLength] = TotalContentLength.ToString(UsCulture);
```
Comment (Bond-009) [inline Emby.Server.Implementations/HttpServer/FileWriter.cs]: Why not use `CultureInfo.InvariantCulture`?

File: Emby.Server.Implementations/HttpServer/FileWriter.cs [hunk 2]
```diff
@@ -99,10 +100,13 @@ private void SetRangeValues()
             RangeStart = requestedRange.Key;
             RangeLength = 1 + RangeEnd - RangeStart;
 
+            // Content-Length is the length of what we're serving, not the original content
+            var lengthString = RangeLength.ToString(UsCulture);
```
Comment (Bond-009) [inline Emby.Server.Implementations/HttpServer/FileWriter.cs]: Same

File: Emby.Server.Implementations/HttpServer/RangeRequestWriter.cs [hunk 3]
```diff
@@ -96,6 +96,7 @@ private void SetRangeValues(long contentLength)
             RangeStart = requestedRange.Key;
             RangeLength = 1 + RangeEnd - RangeStart;
 
+            Headers[HeaderNames.ContentLength] = RangeLength.ToString(UsCulture);
```
Comment (Bond-009) [inline Emby.Server.Implementations/HttpServer/RangeRequestWriter.cs]: Same

File: Emby.Server.Implementations/Services/ResponseHelper.cs [hunk 4]
```diff
@@ -72,51 +71,42 @@ public static Task WriteToResponse(IResponse response, IRequest request, object
...
 
-            var stream = result as Stream;
-            if (stream != null)
-            {
-                return CopyStream(stream, response.OutputStream);
-            }
-
-            var bytes = result as byte[];
-            if (bytes != null)
+            if (result is byte[] bytes)
```
Comment (Bond-009) [inline Emby.Server.Implementations/Services/ResponseHelper.cs]: Why isn't this in the switch statement?

File: Emby.Server.Implementations/Services/ResponseHelper.cs [hunk 5]
```diff
@@ -72,51 +71,42 @@ public static Task WriteToResponse(IResponse response, IRequest request, object
...
                 {
                     return response.OutputStream.WriteAsync(bytes, 0, bytes.Length, cancellationToken);
                 }
+
                 return Task.CompletedTask;
             }
 
-            var responseText = result as string;
-            if (responseText != null)
+            if (result is string responseText)
```
Comment (Bond-009) [inline Emby.Server.Implementations/Services/ResponseHelper.cs]: Same

File: MediaBrowser.Api/Playback/Progressive/BaseProgressiveStreamingService.cs [hunk 6]
```diff
@@ -317,6 +319,12 @@ private async Task<object> GetStaticRemoteStreamResult(StreamState state, Dictio
                 responseHeaders[HeaderNames.AcceptRanges] = "none";
             }
 
+            // Seeing cases of -1 here
+            if (response.ContentLength.HasValue && response.ContentLength.Value >= 0)
+            {
+                responseHeaders[HeaderNames.ContentLength] = response.ContentLength.Value.ToString(UsCulture);
```
Comment (Bond-009) [inline MediaBrowser.Api/Playback/Progressive/BaseProgressiveStreamingService.cs]: Same as above

File: MediaBrowser.Api/Playback/Progressive/BaseProgressiveStreamingService.cs [hunk 7]
```diff
@@ -356,10 +364,34 @@ private async Task<object> GetStreamResult(StreamRequest request, StreamState st
             var contentType = state.GetMimeType(outputPath);
 
             // TODO: The isHeadRequest is only here because ServiceStack will add Content-Length=0 to the response
+            var contentLength = state.EstimateContentLength || isHeadRequest ? GetEstimatedContentLength(state) : null;
+
+            if (contentLength.HasValue)
+            {
+                responseHeaders[HeaderNames.ContentLength] = contentLength.Value.ToString(UsCulture);
```
Comment (Bond-009) [inline MediaBrowser.Api/Playback/Progressive/BaseProgressiveStreamingService.cs]: Same

File: MediaBrowser.Api/Playback/Progressive/BaseProgressiveStreamingService.cs [hunk 8]
```diff
@@ -356,10 +364,34 @@ private async Task<object> GetStreamResult(StreamRequest request, StreamState st
...
+
+                if (streamResult is IHasHeaders hasHeaders)
+                {
+                    if (contentLength.HasValue)
+                    {
+                        hasHeaders.Headers[HeaderNames.ContentLength] = contentLength.Value.ToString(CultureInfo.InvariantCulture);
+                    }
+                    else
+                    {
+                        if (hasHeaders.Headers.ContainsKey(HeaderNames.ContentLength))
```
Comment (Bond-009) [inline MediaBrowser.Api/Playback/Progressive/BaseProgressiveStreamingService.cs]: Merge else and if?

File: Emby.Server.Implementations/Services/ResponseHelper.cs (hunk 5 above)
Comment (cvium) [reply Emby.Server.Implementations/Services/ResponseHelper.cs]: It depends on `bytes` var

File: MediaBrowser.Api/Playback/Progressive/BaseProgressiveStreamingService.cs [hunk 9]
```diff
@@ -281,10 +281,9 @@ protected async Task<object> ProcessRequest(StreamRequest request, bool isHeadRe
         /// <returns>Task{System.Object}.</returns>
         private async Task<object> GetStaticRemoteStreamResult(StreamState state, Dictionary<string, string> responseHeaders, bool isHeadRequest, CancellationTokenSource cancellationTokenSource)
         {
-            string useragent = null;
-            state.RemoteHttpHeaders.TryGetValue("User-Agent", out useragent);
+            state.RemoteHttpHeaders.TryGetValue(HeaderNames.UserAgent, out var useragent);
 
-            var trySupportSeek = false;
+            const bool trySupportSeek = false;
```
Comment (Bond-009) [inline MediaBrowser.Api/Playback/Progressive/BaseProgressiveStreamingService.cs]: Consts should be PascalCase

File: Emby.Server.Implementations/Services/ResponseHelper.cs (hunk 5 above)
Comment (Bond-009) [reply Emby.Server.Implementations/Services/ResponseHelper.cs]: It doesn't need to ;)

File: MediaBrowser.Api/Playback/Progressive/BaseProgressiveStreamingService.cs (hunk 9 above)
Comment (cvium) [reply MediaBrowser.Api/Playback/Progressive/BaseProgressiveStreamingService.cs]: Not configured in the editorconfig

Comment (Bond-009) [reply MediaBrowser.Api/Playback/Progressive/BaseProgressiveStreamingService.cs]: Rider is basing that on local consts currently in the codebase, right?
I'd have to look, but I'm pretty sure analyzers want PascalCase.

File: MediaBrowser.Api/Playback/Progressive/BaseProgressiveStreamingService.cs [hunk 10]
```diff
@@ -356,10 +363,31 @@ private async Task<object> GetStreamResult(StreamRequest request, StreamState st
...
-                return ResultFactory.GetResult(null, Array.Empty<byte>(), contentType, responseHeaders);
+                var streamResult = ResultFactory.GetResult(null, Array.Empty<byte>(), contentType, responseHeaders);
+
+                if (streamResult is IHasHeaders hasHeaders)
+                {
+                    if (contentLength.HasValue)
+                    {
+                        hasHeaders.Headers[HeaderNames.ContentLength] = contentLength.Value.ToString(CultureInfo.InvariantCulture);
+                    }
+                    else if (hasHeaders.Headers.ContainsKey(HeaderNames.ContentLength))
```
Comment (Bond-009) [inline MediaBrowser.Api/Playback/Progressive/BaseProgressiveStreamingService.cs]: You don't need to check if it exists before removing it. (Unless it's some custom improperly implemented dictionary.

**Pull request merged**